_observing-your-data/trace/distributed-tracing.md

```diff
@@ -0,0 +1,159 @@
+
+```bash
+export OPENSEARCH_JAVA_OPTS="-Dopensearch.experimental.feature.telemetry.enabled=true"
+ ./bin/opensearch
+```
+{% include copy-curl.html %}
+
+### Enable with Docker containers
+
+If you’re running Docker, add the following line to `docker-compose.yml` underneath the `opensearch-node` and environment section:
+
+```bash
+OPENSEARCH_JAVA_OPTS="-Dopensearch.experimental.feature.telemetry.enabled=true"
+```
+{% include copy-curl.html %}
+
+### Enable for OpenSearch development
+
+To enable the distributed feature, you must first enable these features by adding the correct properties to `run.gradle` before building OpenSearch. See the [Developer Guide](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#gradle-build) for information about to use how Gradle to build OpenSearch.
+
+Add the following properties to `run.gradle` to enable the feature:
+
+```bash
+testClusters {
+  runTask {
+    testDistribution = 'archive'
+ if (numZones > 1) numberOfZones = numZones
+    if (numNodes > 1) numberOfNodes = numNodes
+    systemProperty 'opensearch.experimental.feature.telemetry.enabled', 'true'
+ }
+ }
+ ```
+ {% include copy-curl.html %}
+
+### Enable distributed tracing
+
+Once you've enabled the feature flag, you can enable the tracer (which is disabled by default) using the following dynamic setting. This setting can be adjusted dynamically to enable or disable tracing in the running cluster:
+
+```bash
+telemetry.tracer.enabled=true
+```
+{% include copy-curl.html %}
+
+## Install the OpenSearch OpenTelemetry plugin
+
+OpenSearch's distributed tracing framework aims to support various telemetry solutions through plugins. The OpenSearch OpenTelemetry plugin `telemetry-otel` is available and must be installed to enable tracing. The following guide provides you with the installation instructions.
+
+### Exporters
+
+At the moment, the distributed tracing feature generates traces and spans for HTTP requests and subset of transport requests. These traces and spans are initially kept in memory using the OpenTelemetry `BatchSpanProcessor` and then are sent to an exporter based on configured settings. The following are the key components:
+
+1. **Span processors:** As spans conclude on the request path, OpenTelemetry provides them to the `SpanProcessor` for processing and exporting. OpenSearch's distributed tracing framework uses the `BatchSpanProcessor`, which batches spans for specific configurable intervals and then sends them to the exporter. The following configurations are available for the `BatchSpanProcessor`:
+    - `telemetry.otel.tracer.exporter.max_queue_size`: Defines the maximum queue size. When the queue reaches this value, it will be written to the exporter. Default is `2048`. 
+    - `telemetry.otel.tracer.exporter.delay`: Defines the delay. If there are not enough spans to fill the `max_queue_size` until this delay time, they will be flushed. Default is `2 seconds`.
```

Comment:
Please confirm that this is the intended meaning. Also, if you use code font, either use `2` (and add ", in seconds" to the description) or `2s`. Alternatively, just leave it in normal font.
```suggestion
    - `telemetry.otel.tracer.exporter.delay`: Defines the delay---a time period after which spans in the queue will be flushed, even if there are not enough spans to fill the `max_queue_size`. Default is `2 seconds`.
```